homework_1/task_2/progressive_tax/progressiveTax.go

```diff
@@ -0,0 +1,80 @@
+package tax
+
+import (
+	"bufio"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type TaxBracket struct {
+	LowerLimitInclusive float64
+	UpperLimitExclusive float64
+	Percentage float64
+}
+
+func CalculateTax(inputValue float64, taxBrackets []TaxBracket) (float64, error) {
```

Comment:
What if `taxBrackets` is completely broken?
E.g. the brackets are not ordered nicely, one bracket has upper limit lesser than the lower limit etc...